apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx

```diff
@@ -1,19 +1,20 @@
 import { Dispatch, SetStateAction, useEffect } from 'react'
 
-import { HookDappBase, HookDappIframe, HookDappType } from '../../../types/hooks'
+import {
+  HOOK_DAPP_ID_LENGTH,
+  HookDappBase,
+  HookDappType,
+  HookDappWalletCompatibility,
+} from '@cowprotocol/hook-dapp-lib'
+import { useWalletInfo } from '@cowprotocol/wallet'
 
-interface HookDappConditions {
-  position?: 'post' | 'pre'
-  smartContractWalletSupported?: boolean
-}
+import { HookDappIframe } from '../../../types/hooks'
 
-type HookDappBaseInfo = Omit<HookDappBase, 'type'>
+type HookDappBaseInfo = Omit<HookDappBase, 'type' | 'conditions'>
 
-type HookDappManifest = HookDappBaseInfo & {
-  conditions?: HookDappConditions
-}
+const MANDATORY_DAPP_FIELDS: (keyof HookDappBaseInfo)[] = ['id', 'name', 'image', 'version', 'website']
```

Comment:
Thinking about this now.
What happens when there's a duplicate id or name?
Do we have a registry somewhere?

Reply:
Ah, I found the registry file.